[Nigel Lim Xi Tong] iP

nigel27022001 opened the pull request:
# Pengu
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes)

Pengu solves the problem of remembering your tasks. Its
* ~~FAST~~ SUPER FAST to use
* Text-based
* Easy for anyone to use

All you need to do is,

1. Download **Pengu** from [here](https://github.com/nigel27022001/ip/releases/tag/A-Jar)
2. Double click it
3. Add your tasks using the command "todo"/"deadline"/"event"
4. Make sure to use the appropriate syntax
5. Tasks managed! 👍 

And it is ***FREE***
Features:
- [x] Adding Tasks such as Deadline, Event and ToDo
- [x] Managing Tasks
- [ ]  Managing Deadlines(Soon)
- [ ] Reminders

---

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

nigel27022001 committed:
level-0: Rename, Greet, Exit

nigel27022001 committed:
level-0:Rename, Greet, Exit

nigel27022001 committed:
level-1: Echo

nigel27022001 committed:
level-2: Add, List

nigel27022001 committed:
level-3: Mark as Done

nigel27022001 committed:
level-4: ToDos, Events, Deadlines

nigel27022001 committed:
fixed bugs with Events and Deadlines

nigel27022001 committed:
A-TextUiTesting and fixed bugs

nigel27022001 committed:
Merge branch 'master' of github.com:nigel27022001/ip

nigel27022001 committed:
level-5: Error Handling

nigel27022001 committed:
level-6: Delete

nigel27022001 committed:
fixed some bugs and added more exceptions

nigel27022001 committed:
fixed A-TextUiTesting

nigel27022001 committed:
Changed task class to abstract

nigel27022001 committed:
Add saving of task list in the hard disk

nigel27022001 committed:
Fixed some bugs

nigel27022001 committed:
Fixed Some Bugs regarding TaskList

nigel27022001 committed:
Fixed Some Bugs in Storage

nigel27022001 committed:
Merge with branch 'branch-level-7'

nigel27022001 committed:
Fix Branching

nigel27022001 committed:
Merge with branch 'branch-level-7' to fix branch

nigel27022001 committed:
Add Java LocalDateTime functionality

nigel27022001 committed:
Merge master with branch 'branch-level-8'

nigel27022001 committed:
Add additional OOP classes

nigel27022001 committed:
Fix branch problem

nigel27022001 committed:
Add packaging of class in Duke

nigel27022001 committed:
Merge with branch 'add-gradle-support'

nigel27022001 committed:
Add gradle support to Duke

nigel27022001 committed:
Add JUnit tests

nigel27022001 committed:
Build JAR file

nigel27022001 committed:
Fix bug regarding exception

nigel27022001 committed:
Add JavaDoc for Duke Chatbot

nigel27022001 committed:
Add Java Coding Standard alignment

nigel27022001 committed:
Add FindTask capabilities

nigel27022001 committed:
Merge branch 'branch-A-CodingStandard'

nigel27022001 committed:
Merge branch 'branch-A-JavaDoc'

nigel27022001 committed:
Add JavaDoc for FindTaskCommand methods

nigel27022001 committed:
Add checkstyle config

nigel27022001 committed:
Merge branch 'branch-A-CheckStyle'

taniatsq commented on src/main/java/duke/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,51 @@
+     *
+     * @return Boolean value indicating whether this command is an exist command.
+     */
+    public boolean isExit() {
```
Clear and effective way of using isExit() to signal exit command of the application instead of creating another boolean value for exit command.

taniatsq commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,57 @@
+ * Class that represents a Task
+ */
+public abstract class Task {
+    /** String containing the description of the Task. */
```
Opening /** should be on a separate line. Single line comment should follow something like:
```
/**
 * String containing the description of the Task.
 */
 ```

or can just use // for single line comments

taniatsq commented on src/main/java/duke/utility/Storage.java:
```diff
@@ -0,0 +1,73 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
```
Be more consistent with order of import statements. Put ```import java.io```s together and ```import java.util```s together.

taniatsq reviewed (commented):
Nicely done! Logic flow and structure of inherited class are easily understandable. Good job! :)

Just a few comments on javadocs and import statements haha!

nigel27022001 committed:
Add GUI for Pengu

nigel27022001 committed:
Merge branch 'branch-level-10'

casaarlai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,167 @@
+     * @throws DukeException
+     * @throws IOException
+     */
+    public Duke(String FilePath) throws DukeException, IOException {
```
Should the parameter `FilePath` here be in camelCase instead of PascalCase? Specifically `filePath`?

casaarlai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,167 @@
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/PenguUser.png"));
+    private Image pengu = new Image(this.getClass().getResourceAsStream("/images/PenguBot.png"));
```
Could the names for the images here be more descriptive `userImage/penguImage`? It is easy to tell that they are images here. But not sure I like that further away, it is more difficult to tell they are images.
 
Also the naming here differs from src/main/java/duke/MainWindow.java, do you think this is potentially misleading?

casaarlai commented:
I appreciate that methods are named in an extremely self-explanatory manner, makes reading and then understanding your code very easy 😊

casaarlai commented on src/test/java/duke/command/DeleteTaskCommandTest.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    @Test
+    public void execute_validTaskIndex_success() throws DukeException, IOException{
```
I really like the way you name your test methods

nigel27022001 committed:
Add Assertions to Parser Class

nigel27022001 committed:
Improve Code Quality

The parser class has methods that are too long.

Extracting the common behaviours into another method, such that the main
methods are shorter.

nigel27022001 committed:
Merge pull request #1 from nigel27022001/branch-A-Assertions

Add Assertions to Parser Class

nigel27022001 committed:
Merge branch 'master' of github.com:nigel27022001/ip

nigel27022001 committed:
Merge branch 'branch-A-CodeQuality'

nigel27022001 committed:
Add Update Task Functionality

Task information and dates could not be altered after creation, other
than through deleting and adding a new one.

Add a new Command for updating tasks and new methods within the Task
sub-classes that enable the functionality of updating.

Abstraction would be the best way forward, by abstracting the update
task command, allowing for easier alterations and debugging.

nigel27022001 committed:
Add Update Task Functionality

Task information and dates could not be altered after creation, other
than through deleting and adding a new one.

Add a new Command for updating tasks and new methods within the Task
sub-classes that enable the functionality of updating.

Abstraction would be the best way forward, by abstracting the update
task command, allowing for easier alterations and debugging.

nigel27022001 committed:
Merge branch 'branch-C-Update'

nigel27022001 committed:
Add User Guide and Fix bug regarding mark

nigel27022001 committed:
Add User Guide and Fix Bug regarding mark

nigel27022001 committed:
Shorten Parser methods

nigel27022001 committed:
Fix Java Version Compatibility
